### PR TITLE
Hotfix – assets and tags

### PR DESCRIPTION
**Pull request by rimi-itk**
This pull request is created to be able to comment on the code changes when compared til `master`.

### CODE REVIEW

File: src/Bpi/ApiBundle/Domain/Aggregate/Node.php [hunk 1]
```diff
@@ -334,24 +353,250 @@ public function setCtime($ctime)
...
+
+    /**
+     * Get params
+     *
+     * @return Bpi\ApiBundle\Domain\Aggregate\Params $params
+     */
+    public function getParams()
+    {
+        return $this->params;
+    }
```
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Domain/Aggregate/Node.php]: Is this related to “assets” or “tags”?

File: src/Bpi/ApiBundle/Domain/Entity/Author.php [hunk 2]
```diff
@@ -102,4 +102,60 @@ public function transform(Document $document)
...
+
+    /**
+     * Get lastname
+     *
+     * @return string $lastname
+     */
+    public function getLastname()
+    {
+        return $this->lastname;
+    }
```
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Domain/Entity/Author.php:160]: Is this related to “assets” or “tags”?

File: src/Bpi/ApiBundle/Domain/Repository/FileRepository.php [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * FileRepository
+ *
+ * This class was generated by the Doctrine ORM. Add your own custom
+ * repository methods below.
+ */
+class FileRepository extends DocumentRepository
+{
+}
```
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Domain/Repository/FileRepository.php]: Missing newline.

File: src/Bpi/ApiBundle/Resources/config/doctrine/ValueObject.Assets.Name.mongodb .yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Bpi\ApiBundle\Domain\ValueObject\Assets\Height:
+  type: embeddedDocument
+  fields:
+    Height:
```
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Resources/config/doctrine/ValueObject.Assets.Name.mongodb .yml]: This should be `name`, right?

File: src/Bpi/ApiBundle/Transform/Extractor/Params.php [hunk 5]
```diff
@@ -31,6 +31,9 @@ public function __construct(Document $doc)
      */
     public function extract()
     {
+        echo "<pre>";
+        print_r(1);
+        die();
```
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Transform/Extractor/Params.php]: This looks like leftovers from a debugging session …

**Review by rimi-itk — COMMENTED** [review]
@guddo, I've added a few comments on the code. Please address these.

**Comment by OIOIOOOI** [comment]
It's hard to say whether this would work out as intended. There are unit tests for the respective bundle which would outline possible issues, yet I doubt they are working now or have been adjusted to meet the respective changes.

**Comment by rimi-itk** [comment]
@AnatolyMuntean, I agree with your thoughts, but we need to move forward with this and will perform extensive manual testing.

Can you (or someone else) address the comments I've added?

**Comment by OIOIOOOI** [comment]
Okay, cool. I will revise this.

**Comment by rimi-itk** [comment]
From our point of view this is very urgent as we need these changes/fixes to move forward on another project, so the sooner the better for us.

File: src/Bpi/ApiBundle/Domain/Aggregate/Node.php (hunk 1 above)
Comment (OIOIOOOI) [reply src/Bpi/ApiBundle/Domain/Aggregate/Node.php]: Can't say. Looks like auto-generated methods.

File: src/Bpi/ApiBundle/Domain/Entity/Author.php (hunk 2 above)
Comment (OIOIOOOI) [reply src/Bpi/ApiBundle/Domain/Entity/Author.php:160]: Can't say. Looks like auto-generated methods.

File: src/Bpi/ApiBundle/Resources/config/doctrine/ValueObject.Assets.Name.mongodb .yml (hunk 4 above)
Comment (OIOIOOOI) [reply src/Bpi/ApiBundle/Resources/config/doctrine/ValueObject.Assets.Name.mongodb .yml]: Exactly.

**Comment by rimi-itk** [comment]
Thanks, @AnatolyMuntean!

File: app/config/config.yml [hunk 6]
```diff
@@ -113,13 +113,13 @@ doctrine_mongodb:
...
-    script_dir_name: "%kernel.root_dir%/scripts"
-    name: "Bpi APIbundle MongoDB Migrations"
-    namespace: "Bpi\ApiBundle\Migrations\MongoDB"
+#mongo_db_migrations:
+#    collection_name: "migration_versions"
+#    database_name: "bpi"
+#    dir_name: "%kernel.root_dir%/../src/Bpi/ApiBundle/Resources/config/Migrations/MongoDB"
+#    script_dir_name: "%kernel.root_dir%/scripts"
+#    name: "Bpi APIbundle MongoDB Migrations"
+#    namespace: "Bpi\ApiBundle\Migrations\MongoDB"
```
Comment (rimi-itk) [inline app/config/config.yml:122]: Unused code/configuration should be deleted (not just commented out).

**Comment by rimi-itk** [comment]
@AnatolyMuntean, the current work flow (you changing code and me owning the PR and reviewing code) is a mess. Can you checkout the `hotfix` branch, rename and push it, and create a new PR til make our roles clear? Then I will delete this PR to clean thing up.

**Comment by guddo** [comment]
Moved here: https://github.com/inleadmedia/rest-api/pull/25.
